### PR TITLE
dev-python/sphinx: Version bump to 1.5.1

### PR DESCRIPTION
**Pull request by SoapGentoo**
Gentoo-bug: 579106
* EAPI=6
* Basis for ebuild by Brian Dolbec (dol-sen)

Diff:
```
--- sphinx-1.3.5.ebuild
+++ sphinx-1.5.1.ebuild
@@ -1,49 +1,72 @@
-# Copyright 1999-2016 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=5
+EAPI=6
 
-PYTHON_COMPAT=( python2_7 python3_{4,5} pypy{,3} )
+PYTHON_COMPAT=( python2_7 python{3_4,3_5} )
 PYTHON_REQ_USE="threads(+)"
 
 inherit distutils-r1 eutils versionator
 
-MY_PN="Sphinx"
-MY_P="${MY_PN}-${PV}"
-
 DESCRIPTION="Python documentation generator"
 HOMEPAGE="http://sphinx.pocoo.org/ https://pypi.python.org/pypi/Sphinx"
-SRC_URI="mirror://pypi/${MY_PN:0:1}/${MY_PN}/${MY_P}.tar.gz"
+SRC_URI="mirror://pypi/S/${PN^}/${P^}.tar.gz"
 
 LICENSE="BSD"
 SLOT="0"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~ia64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos"
-IUSE="doc latex test"
+IUSE="doc latex net test"
 
 RDEPEND="
 	=dev-python/alabaster-0.7*[${PYTHON_USEDEP}]
 	>=dev-python/docutils-0.11[${PYTHON_USEDEP}]
 	>=dev-python/jinja-2.3[${PYTHON_USEDEP}]
 	>=dev-python/pygments-2.0.1-r1[${PYTHON_USEDEP}]
-	>=dev-python/six-1.4[${PYTHON_USEDEP}]
-	>=dev-python/Babel-1.3[${PYTHON_USEDEP}]
-	!=dev-python/Babel-2.0[${PYTHON_USEDEP}]
+	>=dev-python/six-1.5[${PYTHON_USEDEP}]
+	>=dev-python/Babel-2.1.1[${PYTHON_USEDEP}]
 	>=dev-python/snowballstemmer-1.1[${PYTHON_USEDEP}]
 	>=dev-python/sphinx_rtd_theme-0.1[${PYTHON_USEDEP}]
 	<dev-python/sphinx_rtd_theme-2.0[${PYTHON_USEDEP}]
+	dev-python/imagesize[${PYTHON_USEDEP}]
 	latex? (
 		dev-texlive/texlive-latexextra
 		app-text/dvipng
+	)
+	net? (
+		>=dev-python/requests-2.4.0[${PYTHON_USEDEP}]
+		>=dev-python/sqlalchemy-0.9[${PYTHON_USEDEP}]
+		>=dev-python/whoosh-2.0[${PYTHON_USEDEP}]
 	)"
 DEPEND="${RDEPEND}
 	dev-python/setuptools[${PYTHON_USEDEP}]
 	test? (
 		dev-python/nose[${PYTHON_USEDEP}]
-		$(python_gen_cond_dep 'dev-python/mock[${PYTHON_USEDEP}]' python2_7 pypy)
+		dev-python/mock[${PYTHON_USEDEP}]
+		$(python_gen_cond_dep 'dev-python/simplejson[${PYTHON_USEDEP}]' pypy)
+		dev-python/html5lib[${PYTHON_USEDEP}]
+		>=dev-python/sqlalchemy-0.9[${PYTHON_USEDEP}]
+		>=dev-python/whoosh-2.0[${PYTHON_USEDEP}]
+		dev-python/flake8[${PYTHON_USEDEP}]
+		dev-python/requests[${PYTHON_USEDEP}]
+		$(python_gen_cond_dep 'dev-python/enum34[${PYTHON_USEDEP}]' 'pypy*' 'python2*')
+		dev-python/typing[${PYTHON_USEDEP}]
 	)"
 
-S="${WORKDIR}/${MY_P}"
+S="${WORKDIR}/${P^}"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.5.1-fix-pycode-grammar.patch
+)
+
+python_prepare_all() {
+	# remove tests that fail due to network-sandbox
+	rm tests/test_websupport.py || die "Failed to remove web tests"
+	rm tests/test_searchadapters.py || die "Failed to remove web tests"
+	rm tests/test_build_linkcheck.py || die "Failed to remove web tests"
+
+	distutils-r1_python_prepare_all
+}
 
 python_compile() {
 	distutils-r1_python_compile
@@ -52,85 +75,40 @@
 	# Note that the tests usually do it for us. However, I don't want
 	# to trust USE=test really running all the tests, especially
 	# with FEATURES=test-fail-continue.
-	pushd "${BUILD_DIR}"/lib > /dev/null || die
-	"${PYTHON}" -m sphinx.pycode.__init__ \
-		|| die "Grammar generation failed."
-	popd > /dev/null || die
+	pushd "${BUILD_DIR}"/lib >/dev/null || die
+	"${EPYTHON}" -m sphinx.pycode.__init__ || die "Grammar generation failed."
+	popd >/dev/null || die
 }
 
 python_compile_all() {
-	use doc && emake -C doc SPHINXBUILD='"${PYTHON}" "${S}/sphinx-build.py"' html
+	if use doc; then
+		emake -C doc SPHINXBUILD='"${EPYTHON}" "${S}/sphinx-build.py"' html
+		HTML_DOCS=( doc/_build/html/. )
+	fi
 }
 
 python_test() {
 	mkdir -p "${BUILD_DIR}/sphinx_tempdir" || die
-	export SPHINX_TEST_TEMPDIR="${BUILD_DIR}/sphinx_tempdir"
-	cp -r -l tests "${BUILD_DIR}"/ || die
-
-	if $(python_is_python3); then
-		2to3 -w --no-diffs "${BUILD_DIR}"/tests || die
-	fi
-
-	nosetests -w "${BUILD_DIR}"/tests -v \
-		|| die "Tests fail with ${EPYTHON}"
-}
-
-python_install_all() {
-	use doc && local HTML_DOCS=( doc/_build/html/. )
-
-	distutils-r1_python_install_all
-}
-
-replacing_python_eclass() {
-	local pv
-	for pv in ${REPLACING_VERSIONS}; do
-		if ! version_is_at_least 1.1.3-r4 ${pv}; then
-			return 0
-		fi
-	done
-
-	return 1
-}
-
-pkg_preinst() {
-	if replacing_python_eclass; then
-		# the old python.eclass ebuild will want to remove our pickles...
-		backup_pickle() {
-			# array to enable filename expansion
-			local pickle_name=(
-				"${D}$(python_get_sitedir)"/sphinx/pycode/Grammar*.pickle
-			)
-
-			local dest=${ROOT}${pickle_name[0]#${D}}.backup
-
-			eumask_push 022
-			mkdir -p "${dest%/*}" || die
-			eumask_pop
-
-			cp -p -v "${pickle_name[0]}" "${dest}" \
-				|| die "Unable to backup grammar pickle from overwriting"
-		}
-
-		python_foreach_impl backup_pickle
-	fi
+	local -x SPHINX_TEST_TEMPDIR="${BUILD_DIR}/sphinx_tempdir"
+	cp -r -l tests "${BUILD_DIR}"/ || die "Failed to copy tests"
+	cp Makefile "${BUILD_DIR}"/ || die "Failed to copy Makefile"
+	emake test
 }
 
 pkg_postinst() {
-	if replacing_python_eclass; then
-		local warned
-
-		restore_pickle() {
-			local backup_name=(
-				"${ROOT}$(python_get_sitedir)"/sphinx/pycode/Grammar*.pickle.backup
-			)
-			local dest=${backup_name[0]%.backup}
+	replacing_python_eclass() {
+		local pv
+		for pv in ${REPLACING_VERSIONS}; do
+			if ! version_is_at_least 1.1.3-r4 ${pv}; then
+				return 0
+			fi
+		done
 
-			mv -v "${backup_name[0]}" "${dest}" \
-				|| die "Unable to restore grammar pickle backup"
-		}
+		return 1
+	}
 
-		python_foreach_impl restore_pickle
-
-		[[ ${warned} ]] && ewarn "Please try rebuilding the package."
+	if replacing_python_eclass; then
+		ewarn "Replaced a very old sphinx version. If you are"
+		ewarn "experiencing problems, please re-emerge sphinx."
 	fi
 }
```

### CODE REVIEW

**Comment by SoapGentoo** [comment]
@floppym @dol-sen 
- Am I seeing this correctly, `USE="latex"` sounds very automagic-ish to me. What Gentoo mechanism triggers generating/utilizing latex?
- Can we remove that legacy backup code at some point?

**Comment by dol-sen** [comment]
To be honest, I am not familiar with using sphinx.  I originally started on the 1.4 bump because of a newer buildbot reqirement.  I just went through setup.py requirements along with some additional test deps from looking at the tox.ini and travis requirements.

from setup.py in their git repo:
* Output formats: HTML (including derivative formats such as HTML Help, Epub
  and Qt Help), plain text, manual pages and LaTeX or direct PDF output
  using rst2pdf

But it is not listing any latex deps in setup.py requirements.  Nor are there any in tox.ini

But, in travis.yml (which shows a commit only 2 hours ago, so still used):

addons:
  apt:
    packages:
    - graphviz
    - texlive-latex-recommended
    - texlive-latex-extra
    - texlive-fonts-recommended
    - texlive-fonts-extra
    - texlive-xetex
    - lmodern

File: dev-python/sphinx/files/sphinx-1.5.1-fix-pycode-grammar.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+environment.py has been turned into a directory,
+without actually porting the code using it.
```
Comment (floppym) [inline dev-python/sphinx/files/sphinx-1.5.1-fix-pycode-grammar.patch]: Please send this patch upstream if it has not been done already.

File: dev-python/sphinx/sphinx-1.5.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+inherit distutils-r1 eutils versionator
+
+DESCRIPTION="Python documentation generator"
+HOMEPAGE="http://sphinx.pocoo.org/ https://pypi.python.org/pypi/Sphinx"
+SRC_URI="mirror://pypi/S/${PN^}/${P^}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (floppym) [inline dev-python/sphinx/sphinx-1.5.1.ebuild]: Why did the other keywords get dropped?

**Comment by floppym** [comment]
> Am I seeing this correctly, USE="latex" sounds very automagic-ish to me. What Gentoo mechanism triggers generating/utilizing latex?

I assume it is there so some other package is hard-coded to produce latex output and depends on dev-python/sphinx[latex]. I don't know what package that is or if it still exists.

> Can we remove that legacy backup code at some point?

Yes, go ahead.

**Pull request closed without merging**